### ISSUE TITLE
Fix simple_fixture and never_skip compatibility

### ISSUE THREAD

**Issue by adler-j** [odlgroup/odl#1149]
These are currently somewhat in-compatible and code like this is needed:

```python
fft_impl = simple_fixture('fft_impl',
                          [odl.util.testutils.never_skip('numpy'),
                           odl.util.testutils.skip_if_no_pyfftw('pyfftw')],
                          fmt=" {name} = '{value.args[1]}' ")
```

**Comment by kohr-h** [comment]
About the first part I'm not too unhappy, just the `'fmt'` magic could use a simple check for avoiding this. We would need sth like this:
```python
def simple_fixture(name, params, fmt=None):
    import _pytest
    if fmt is None:
        fmt_str = " {name} = '{value}' "
        fmt_skipif = " {name} = {value.args[1]} "
        fmt_skipif_str = " {name} = '{value.args[1]}' "
        fmt_default = " {name} = {value} "

    ids = []
    for p in params:
        if fmt is None:
            if isinstance(p, _pytest.mark.MarkDecorator) and p.name == 'skipif':
                try:
                    p.args[1] + ''
                except TypeError:
                    ids.append(fmt_skipif.format(p.args[1]))
                else:
                    ids.append(fmt_skipif_str.format(p.args[1]))
            else:
                try:
                    p + ''
                except TypeError:
                    ids.append(fmt_default.format(p))
                else:
                    ids.append(fmt_str.format(p))
        else:
            ids.append(fmt.format(p))

    # Rest of the function
```

**Comment by kohr-h** [comment]
Closed by #1154
